plugins/okta/pkg/okta/okta.go

```diff
@@ -343,6 +370,52 @@ func (oktaPlugin *Plugin) Extract(req sdk.ExtractRequest, evt sdk.EventReader) e
 				req.SetValue(i.DisplayName)
 			}
 		}
+	case "okta.mfa.failure.countlast":
+		if data.EventType == "user.authentication.auth_via_mfa" && data.Outcome.Result == "FAILURE" {
+			key := data.EventType + ":" + data.Actor.ID
+			shift, err := strconv.Atoi(req.ArgKey())
+			var count uint64
+			if err != nil {
+				return fmt.Errorf("bad argument %v for field %v", req.ArgKey(), req.Field())
+			}
+			value, err := oktaPlugin.cache.Get(key)
+			if err == nil {
+				if value != nil {
+					values := strings.Split(value.(string), ",")
+					oldest := time.Now().Add(-time.Duration(shift) * time.Second).UnixNano()
+					for _, i := range values {
+						v, _ := strconv.ParseUint(i, 10, 64)
+						if v > uint64(oldest) {
+							count++
+						}
+					}
+				}
+			}
+			req.SetValue(count)
```

Comment:
I think there's an issue here.

This code path is entered each time the field is extracted. The engine calls the field extraction each time it's present in a rule condition or in an output field.

So for example, if two rules use `okta.mfa.failure.countlast`, the counter will be incremented twice each round.
That's not the expected behavior, I guess.

Reply:
I think it's ok, the global count of events is set in cache and this cache is updated only once, at first extraction, see https://github.com/falcosecurity/plugins/pull/156/files#diff-9980b0fcfd1006e1a8087ec9ea20e4cc6f80d1a63dc15da73c2bb5126b497303R250
When we want to count the number of events in last X seconds, we get the cache and increment the `count` variable for each element after (now - X seconds).